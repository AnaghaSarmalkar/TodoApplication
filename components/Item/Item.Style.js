import { StyleSheet } from "react-native";

export const style = StyleSheet.create({
  item: {
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
  },
  name: {
    fontSize: 32,
  },
});
