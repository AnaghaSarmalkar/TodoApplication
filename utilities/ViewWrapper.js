import React from "react";
import { View } from "react-native";

export default function ViewWrapper(props) {
  return (
    <View
      {...props}
      // style={{
      //   flex: 1,
      //   alignItems: "center",
      //   justifyContent: "center",
      //   // paddingTop: insets.top,
      // }}
    >
      {props.children}
    </View>
  );
}
