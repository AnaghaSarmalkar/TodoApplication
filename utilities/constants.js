import { Platform } from "react-native";

export const isIos = Platform.OS === "ios";
export const isWeb = Platform.OS === "web";
export const isAndroid = Platform.OS === "android";
