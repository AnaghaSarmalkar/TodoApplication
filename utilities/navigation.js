export function setPageTitle(navigation, title) {
  if (title) {
    navigation.setOptions({
      title: title,
    });
  }
}
