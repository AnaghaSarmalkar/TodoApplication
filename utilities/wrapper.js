import React from "react";
import { SafeAreaView } from "react-native-safe-area-context";

function SafeAreaWrapper({ children }) {
  <SafeAreaView>{children}</SafeAreaView>;
}
