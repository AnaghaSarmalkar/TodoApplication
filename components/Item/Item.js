import React from "react";
import { Text, TouchableOpacity } from "react-native";
import { style } from "./Item.Style";

export default function Item({ item }) {
  return (
    <TouchableOpacity style={[style.item]}>
      <Text style={[style.name]}>{item.name}</Text>
    </TouchableOpacity>
  );
}
