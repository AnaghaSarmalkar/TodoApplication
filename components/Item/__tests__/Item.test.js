import React from "react";
import { shallow } from "../../../utilities/testing";
import Item from "../Item";

describe("", () => {
  it("renders Item component", async () => {
    const item = { id: "1", name: "Test Item name" };
    const wrapper = shallow(<Item item={item} onItemPress={undefined} />);

    expect(wrapper).toMatchSnapshot();
  });
});
