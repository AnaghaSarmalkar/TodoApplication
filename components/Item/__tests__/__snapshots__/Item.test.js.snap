// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders Item component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "padding": 10,
    }
  }
>
  <Memo(Pressable)
    onPress={[Function]}
    style={
      [
        {
          "marginHorizontal": 16,
          "marginVertical": 8,
          "padding": 20,
        },
      ]
    }
  >
    <View>
      <Text
        style={
          [
            {
              "fontSize": 32,
            },
          ]
        }
      >
        Test Item name
      </Text>
    </View>
  </Memo(Pressable)>
</View>
`;
